### DISCUSSION TITLE
Header

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#160] [Feedback]
Use this place to discuss the [Header](https://ons-design-system.netlify.com/components/header) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
The header shouldn't have a margin bottom as it prevents being able to add components such as a hero banner flush to the header. This margin should be added to the [page template](#195) instead.

**Reply by jrbarnes9** [reply]
#203 resolved

**Comment by bameyrick** [comment]
The reference to `/img/ons-logo-black-en.svg` needs to also have a version number parameter for usage in the [prototype kit](https://github.com/ONSdigital/prototype-kit).

**Reply by jrbarnes9** [reply]
#213 resolved

**Comment by jrbarnes9** [comment]
The mobile header nav is missing `display:none` when collapsed.  You can still click the menu links when it's collapsed.

**Reply by jrbarnes9** [reply]
#312 resolved

**Comment by bameyrick** [comment]
The header logo isn't rendering correctly in IE11:

<img width="443" alt="Screenshot 2019-04-18 at 15 15 04" src="https://user-images.githubusercontent.com/8447446/56368302-0de99280-61ef-11e9-8a67-ba639cd58ea9.png">

**Reply by jrbarnes9** [reply]
#315 resolved

**Comment by bameyrick** [comment]
Header navigation doesn't work on mobile without JS

**Reply by jrbarnes9** [reply]
#317 resolved

**Comment by bameyrick** [comment]
The main navigation always sets a homepage as active. This issue has been raised here: https://github.com/ONSdigital/prototype-kit/issues/10

**Reply by jrbarnes9** [reply]
#400 resolved

**Comment by jrbarnes9** [comment]
No padding between title logo and menu button on mobile ~320px...

<img width="445" alt="Screenshot 2019-06-13 at 11 47 07" src="https://user-images.githubusercontent.com/43346934/59426857-29c57b00-8dd1-11e9-9480-72d415337744.png">

**Reply by jrbarnes9** [reply]
#425 resolved

**Comment by bameyrick** [comment]
The logo on the internal variant of the header isn't rendering in any browser:

![Screenshot 2019-06-18 at 10 42 13](https://user-images.githubusercontent.com/8447446/59671629-e8104800-91b5-11e9-837b-e704dbfdff5a.png)

**Reply by jrbarnes9** [reply]
#436 resolved

**Comment by bameyrick** [comment]
@will-madebymoon has requested the min-height of the top bar on the header to be dropped to 45px

**Comment by jrbarnes9** [comment]
Nav links are being hidden from screen readers due to `Aria-hidden=true`

**Reply by jrbarnes9** [reply]
#452 resolved

**Comment by bameyrick** [comment]
Mobile navigation JavaScript has appeared to have stopped working

**Reply by jrbarnes9** [reply]
#554 resolved

**Comment by jrbarnes9** [comment]
![image (3)](https://user-images.githubusercontent.com/43346934/66130264-57853880-e5e9-11e9-92a6-35337b5a74b1.png)

**Reply by jrbarnes9** [reply]
#621 resolved

**Comment by jrbarnes9** [comment]
#756 When the ONS logo is wrapped in a link tag. The link class 'header__logo-link' is using 'inline-block', which causes extra padding. Changing this from 'inline-block' to 'block' should fix the issue.

**Subtle alignment problem**

![image](https://user-images.githubusercontent.com/4989027/82878638-a2b2d980-9f33-11ea-8b8a-a69cbb4751e9.png)

**Fixed using block instead of inline-block**

![image](https://user-images.githubusercontent.com/4989027/82878815-ef96b000-9f33-11ea-8f94-67b544d42945.png)

**Reply by jrbarnes9** [reply]
#756 resolved

**Comment by trevorsaint** [comment]
Census logo needs to be changed to match the branding. 2021 should be white at 70% opacity.

Currently this
![image](https://user-images.githubusercontent.com/4989027/87169263-9a3a1680-c2c7-11ea-8503-b2ff3e4b7d1d.png)

Should be this
![image](https://user-images.githubusercontent.com/4989027/87169278-a3c37e80-c2c7-11ea-87bf-91d5aab7dddf.png)

**Reply by jrbarnes9** [reply]
#830 resolved

**Comment by trevorsaint** [comment]
Alignment of logo and menu button in the header is incorrect.


Adding `display:block` to the SVG image will resolve this and render the header correctly.

**Before**
![before](https://user-images.githubusercontent.com/4989027/94233082-36ef7600-feff-11ea-9a7b-46ea6a37ac91.png)

**After**
![after](https://user-images.githubusercontent.com/4989027/94233103-4373ce80-feff-11ea-87d1-37a79e0a7a25.png)

This also fixes the over-spacing between the logo and the navigation when in desktop view.

**Reply by jrbarnes9** [reply]
#1245 resolved

**Comment by jrbarnes9** [comment]
Missing focus states on masthead and title header logos

**Reply by jrbarnes9** [reply]
#1198 resolved

**Comment by trevorsaint** [comment]
Looking at the source code.

Alt tags should NOT be on the source tag. Only the img tag.

<img width="1093" alt="Screenshot 2020-12-10 at 16 26 51" src="https://user-images.githubusercontent.com/4989027/101799873-8a3b9280-3b04-11eb-93e7-57f6076cdaab.png">

**Reply by jrbarnes9** [reply]
#1230 resolved

**Comment by jrbarnes9** [comment]
The vertical position of the census logo, and the height of the header block are different between a default header, and header with sign out button

**Reply by jrbarnes9** [reply]
#1260 resolved

**Comment by jrbarnes9** [comment]
The width of the census logo should be the same regardless of mobile menu
![Screenshot 2020-12-30 at 11 24 47](https://user-images.githubusercontent.com/43346934/103348694-b0fd4100-4a92-11eb-86d0-82bf7f54abe6.png)
![Screenshot 2020-12-30 at 11 28 51](https://user-images.githubusercontent.com/43346934/103348700-b35f9b00-4a92-11eb-8207-d13896d096dd.png)

**Reply by jrbarnes9** [reply]
#1266 resolved

**Comment by jrbarnes9** [comment]
Remove `tabIndex` from `header__logo` if `logoHref` not defined

Currently, the logo is part of the tab index even when there is no anchor present

**Reply by jrbarnes9** [reply]
#1333 resolved

**Comment by jrbarnes9** [comment]
Header broken in IE11

Traced back to PR #1036 
<img width="1280" alt="Screenshot 2021-01-29 at 09 25 22" src="https://user-images.githubusercontent.com/43346934/106258008-36504e80-6215-11eb-8d54-3426ad54e8ae.png">

**Reply by jrbarnes9** [reply]
#1340 resolved

**Comment by jrbarnes9** [comment]
Service Links missing from internal header example and incorrectly styled when applied (should be white)

**Reply by jrbarnes9** [reply]
#1577 resolved

**Comment by jrbarnes9** [comment]
Header vertical spacing issue when header title wraps

![Screenshot 2021-08-13 at 09 00 15](https://user-images.githubusercontent.com/43346934/129324553-1539fb47-efa9-44a4-86b8-7373073c5dd1.png)
![Screenshot 2021-08-13 at 08 58 24](https://user-images.githubusercontent.com/43346934/129324556-8d5a2f41-1a31-411f-afa4-870d678ceed3.png)

**Reply by jrbarnes9** [reply]
Issue #1636 raised

**Comment by jrbarnes9** [comment]
Header title with button not vertically centre aligned 
![Screenshot 2021-12-13 at 16 25 52](https://user-images.githubusercontent.com/43346934/145850161-59f36202-36cb-43c3-aab2-1a78ce1fa670.png)

**Reply by jrbarnes9** [reply]
Raised #1799

**Comment by jrbarnes9** [comment]
The active item in the header nav should allow for the extra width of the bold font and not cause the subsequent nav items to shift horizontally.

![header-nav-issue](https://user-images.githubusercontent.com/43346934/148212415-76baa723-84bf-45b6-898b-0b687b3d1567.gif)

**Reply by jrbarnes9** [reply]
Raised #1809

**Comment by jrbarnes9** [comment]
We should add all the header parameters to page template so they can be used. 

Missing params: `titleAsH1`, `wide`, `fullWidth`, `description`

**Reply by jrbarnes9** [reply]
Raised #2095

**Comment by jrbarnes9** [comment]
Typo in header macro parameter `tilteAsH1`

Should be `titleAsH1`

**Reply by jrbarnes9** [reply]
Raised #2152

**Comment by jrbarnes9** [comment]
Allow static strings in `params.ServiceLinks` to e.g. show the logged in user name
![image](https://user-images.githubusercontent.com/43346934/175057403-2fe6175d-7a9f-4e7c-b866-63ebb48f400b.png)

**Reply by jrbarnes9** [reply]
Raised #2273

**Comment by jrbarnes9** [comment]
Header doesn't cater for full colour brand logos

**Reply by jrbarnes9** [reply]
Raised #2281

**Comment by jrbarnes9** [comment]
Masthead should be optional for services that don't require the 'org' branding

**Reply by jrbarnes9** [reply]
Raised #2289

**Comment by jrbarnes9** [comment]
DAC: Remove `aria-haspopup` attr from service links menu toggle button

**Reply by jrbarnes9** [reply]
#2333

**Comment by jrbarnes9** [comment]
DAC: Remove `aria-label` from service links `ul`

**Reply by jrbarnes9** [reply]
#2338

**Comment by Delorenzo6000** [comment]
Hey team! We would like you add to the DS the new header for the IDS Hub. Jon Barnes just texted: "I’ve managed to recreate the header in the DS with no changes, just needs the following config. The only change we’ll need to make is the flexible height of the masthead for the mobile logo". The new colour is: #4d076b. Also the size for title "IDS Hub" is: ons-u-fs-xxl (2rem). Thank you

**Reply by jrbarnes9** [reply]
Raised #2431

**Reply by jrbarnes9** [reply]
Raised #2432

**Reply by jrbarnes9** [reply]
Raised #2433

**Comment by Delorenzo6000** [comment]
Hey team! We would like you add to the DS a footer template where the 'gov.uk' and 'ONS sponsorship' logos are a link (it is a requirements for the IDS website and IDS Hub), thank you!

**Reply by jrbarnes9** [reply]
Referenced in Footer discussion: https://github.com/ONSdigital/design-system/discussions/172#discussioncomment-3537081

**Comment by jrbarnes9** [comment]
`fullWidth` param missing for sub navigation.

![Screenshot 2023-02-15 at 11 13 53](https://user-images.githubusercontent.com/43346934/219012392-bdc7926d-15f4-45f9-ad4a-a6721c53082f.png)

**Reply by jrbarnes9** [reply]
Raised #2615

**Comment by Paul-Joel** [comment]
There appears to be a bug in the Service nav. 
If you reduce the width of the header until the Service nav shifts to the 'Account' nav and open it, the vertical nav does not hide itself when the page is made wider again and the horizontal nav version reappears. 

![image](https://github.com/ONSdigital/design-system/assets/19411155/f5a4ad96-6dd3-4da4-94bc-1aadbb9477ce)

**Comment by romaricpascal** [comment]
Doing some research for GOV.UK Frontend around navigation, noticed that the [External with service links example](https://service-manual.ons.gov.uk/design-system/components/header/example-header-external-with-service-links) didn't show any service links when accessed on a narrow viewport (412px when I tested) and JavaScript doesn't load.

![Screenshot 2024-02-01 at 14 15 13](https://github.com/ONSdigital/design-system/assets/396367/9ff55c58-8e57-48ae-8884-cd43aecaddc7)

**Comment by romaricpascal** [comment]
I also noticed a similar thing on the [External with sub navigation example](https://service-manual.ons.gov.uk/design-system/components/header/example-header-external-with-sub-navigation), where the sub navigation doesn't show on narrow viewport (412px) if JavaScript doesn't load.

![Screenshot 2024-02-01 at 14 12 32](https://github.com/ONSdigital/design-system/assets/396367/ae1aa271-9444-4eed-ad8b-0d964321b00f)

**Comment by AlanHONS** [comment]
Currently ONS Header component adds explicit role="banner" to `<header class="ons-header" role="banner">` HTML output.

However that is flagged as a warning of failure against WCAG2 Level A 4.1.2 and listed as something that shouldn't be done as per https://www.w3.org/TR/html-aria/#avoid-specifying-redundant-roles Because the HTML `<header>` tag already has a default Role of Banner. So explicit and duplicate declaration in this case is advised against in this specific case. 
Unless there is some other good reason why that role= attribute was specifically added to `<header>`?

Error was found and flagged by https://www.totalvalidator.com/ testing tool:
<img width="890" alt="image" src="https://github.com/user-attachments/assets/26e4568f-2927-4529-a8f6-976222e4c4c4" />

**Reply by MagdalenaLarge** [reply]
HI @AlanHONS , I will raise this issue with the team and let you know once we have more details. Thank you

**Reply by MagdalenaLarge** [reply]
Hi @AlanHONS we review your request and we created ticket to resolve this issue. You can track progress of it : https://jira.ons.gov.uk/browse/ONSDESYS-529. Please let us know in case of any further questions. Thank you for your patience.

**Comment by AlanHONS** [comment]
Please review current SVG title= attributes associated with mobile version Main menu and Service link button chevron arrows.
As currently these are unfriendly ID style values that could be read out to assistive technology users and has been flagged as warning of failing WCAG2 Level A 4.1.2.

Given that both those SVGs are right next to the text of button names and within a properly coded <button> tag including aria-expanded="true/false" information I believe that those arrows are purely visual presentation elements?
And therefore just have their title= attributes removed? 

If that causes other problems then next best solution might (?) be empty title="", like an IMG empty alt=""?
Otherwise worst case change contents of title= to something human readable and simple plain English like title="Up arrow"?

Evidence as found and flagged by https://www.totalvalidator.com/ testing tool:
![image](https://github.com/user-attachments/assets/d5ab54e1-0644-47ac-8ae4-2d319dfe4c24)

![image](https://github.com/user-attachments/assets/5f93271e-8023-4422-91c6-7d3ab9d28096)

**Reply by MagdalenaLarge** [reply]
Hello @AlanHONS , I have a very good news. The issue you raised above has been resolved as part of the tickets: https://jira.ons.gov.uk/browse/ONSDESYS-465 and  https://jira.ons.gov.uk/browse/ONSDESYS-467. They will be included in our next release (within next 2 weeks). Thank you.

**Comment by sanjeevz3009** [comment]
# Header with search button responsiveness issue

## Mobile Header Misalignment at Small Viewports (≤396px Width)

I've identified a responsiveness issue with the ONS Header component when viewed at small mobile viewport widths (approximately 396px and below, e.g. iPhone SE dimensions in Chrome DevTools).

At these viewport sizes, the navigation controls, specifically the Menu and Search buttons,  become visually misaligned and appear out of place. The layout does not scale cleanly at this breakpoint, resulting in a header that looks broken or unpolished.

This affects internal CMS users (editors, publishing officers, admins) who access the system and look at the preview pane in the Wagtail CMS or external users on smaller devices or reduced-width windows.

## See the pictures below:
### Wagtail preview pane
<img width="1573" height="892" alt="Screenshot 2026-02-26 at 18 01 15" src="https://github.com/user-attachments/assets/b88ba1c0-d8e0-49b7-a5a8-a18e73940cff" />

### External user view of a page
<img width="969" height="822" alt="Screenshot 2026-02-26 at 18 01 28" src="https://github.com/user-attachments/assets/fdc1ade2-e1e1-4218-a8a3-06174f806b76" />

**Reply by MagdalenaLarge** [reply]
Hi @sanjeevz3009 Thank you for raising it up. Can I ask you for the link with the example of how this actually behave on the website not within Wagtail preview? Thank you.
